scripts/cloudPBFShardControl/cloudPBFShardControl.py

```diff
@@ -0,0 +1,608 @@
+
+    parser_sync = subparsers.add_parser(
+        "sync", help="Sync pbf files from instance to S3 folder"
+    )
+    parser_sync.add_argument(
+        "-i",
+        "--id",
+        required=True,
+        help="ID - Indicates the ID of an existing VM instance to use",
+    )
+    parser_sync.add_argument(
+        "-k",
+        "--key",
+        required=True,
+        help="KEY - Instance key name to use to login to instance. This key "
+        "is expected to be the same name as the key as defined by AWS and the "
+        "corresponding pem file must be located in your local '~/.ssh/' "
+        "directory and should be a pem file. See the following URL for "
+        "instructions on creating a key: "
+        "https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html. "
+        "(e.g. `--key=aws-key`)",
+    )
+    parser_sync.add_argument(
+        "-o", "--out", required=True, help="Out - The S3 Output directory"
+    )
+    parser_sync.set_defaults(func=cloudctl.sync)
+
+    parser_clean = subparsers.add_parser("clean", help="Clean up instance")
+    parser_clean.add_argument(
+        "-i",
+        "--id",
+        required=True,
+        help="ID - Indicates the ID of an existing VM instance to use",
+    )
+    parser_clean.add_argument(
+        "-k",
+        "--key",
+        required=True,
+        help="KEY - Instance key name to use to login to instance. This key "
+        "is expected to be the same name as the key as defined by AWS and the "
+        "corresponding pem file must be located in your local '~/.ssh/' "
+        "directory and should be a pem file. See the following URL for "
+        "instructions on creating a key: "
+        "https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html. "
+        "(e.g. `--key=aws-key`)",
+    )
+    parser_clean.set_defaults(func=cloudctl.clean)
+
+    args = parser.parse_args()
+    return args
+
+
+def evaluate(args, cloudctl):
+    """Evaluate the given arguments.
+
+    :param args: The user's input.
+    :param cloudctl: An instance of CloudPBFShardControl to use.
+    """
+    if args.terminate:
+        cloudctl.terminate = args.terminate
+    if args.name:
+        cloudctl.instanceName = args.name
+    if args.template:
+        cloudctl.templateName = args.templateName
+    if args.minutes:
+        cloudctl.timeoutMinutes = args.minutes
+    if hasattr(args, "processes") and args.processes:
+        cloudctl.processes = args.processes
+    if hasattr(args, "key") and args.key:
+        cloudctl.key = args.key
+    if hasattr(args, "out") and args.out:
+        cloudctl.s3Folder = args.out
+    if hasattr(args, "pbf") and args.pbf:
+        cloudctl.pbfURL = args.pbf
+    if hasattr(args, "id") and args.id:
+        cloudctl.instanceId = args.id
+        cloudctl.get_instance_info()
+    if args.version:
+        logger.critical("This is version {0}.".format(VERSION))
+        finish()
```

Comment:
I thought I had put a comment on this in here but I didn't see it...

Something to consider with these checks